interview-questions.md

```diff
@@ -8,62 +8,85 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+Your answer:
+The .map() function iterates through an array and returns a new array of the same length.
 
-  Researched answer:
+Researched answer: 
+
+The .map method have three built-in parameters which are the value, index, and array. The value is the value "required" in the collection at that point in the iteration. The index which is optional - is the position in the collection. Lastly, the array is the entire collection that method is iterating through, which is also optional. 
 
 
 
 3. What is the difference between map and filter?
 
   Your answer:
+  The .map() function iterates through an array and returns a new array of the same length.
+
+  The .filter() function takes an array and filters out unwanted elements.
 
   Researched answer:
+  When you call .map on an array, it executes that callback on every element within it, returning a new array with all of the values that the callback returned.
+
+  Filter function is for defining a new array with elements from the initial array that match a specified condition/function.
 
 
 
 4. What is the difference between a function and a method?
 
   Your answer:
+  A function is a set of instructions detailing how to do a task. When a function belongs to an object, it is called a method.
+  
 
   Researched answer:
+  Functions are reusable pieces of code that only execute when invoked and always return an output.
 
+  Method is a function associated with a particular class and contained within an object. Objects being non-primitve data-type that store mulitple collections of data and values that consists properties.
 
 
 5. What is object destructuring?
 
   Your answer:
+  When destructuring an object, you are essentially taking the properties of an object and breaking them out into individual, recallable variables.
 
   Researched answer:
-
+  Object destructuring makes it possible to unpack values from arrays, or properties from objects, into distinct variables.
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer:
+  I was prior Navy, prior Air Force, and currently in Space Force and have encountered the word hoisting multiple times in my career. Literally, hoisting means raising something to the top. JavaScript hoisting in the other hand, I believe it should be something related to functions putting it on top, or when it is first created it should start somehwere else. before you console log it.
 
   Researched answer:
-
+  Hoisting happens because JavaScript wants to find all the variables and functions before running the code to reserve memory as it compiles them. Before JavaScript code is executed, it is first compiled. 
+  As it is compiled, JavaScript moves all function and variable declarations to the start of the script. This causes hoisting!
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
```

Comment:
Excited that we are bringing in higher concepts to this foundation you have just proven that you have.